examples/platforms/cc2652/radio.c

```diff
@@ -0,0 +1,1861 @@
+
+    sTransmitCmd = cTransmitCmd;
+    /* no need to look for an ack if the tx operation was stopped */
+    sTransmitCmd.payloadLen = aLen;
+    sTransmitCmd.pPayload = aPsdu;
+
+    if (aPsdu[0] & IEEE802154_ACK_REQUEST)
+    {
+        /* setup the receive ack command to follow the tx command */
+        sTransmitCmd.condition.rule = COND_STOP_ON_FALSE;
+        sTransmitCmd.pNextOp = (rfc_radioOp_t *) &sTransmitRxAckCmd;
+
+        sTransmitRxAckCmd = cTransmitRxAckCmd;
+        sTransmitRxAckCmd.seqNo = aPsdu[IEEE802154_DSN_OFFSET];
+    }
+
+    return (RFCDoorbellSendTo((uint32_t)&sCsmacaBackoffCmd) & 0xFF);
+}
+
+/**
+ * Sends the rx command to the radio core.
+ *
+ * Sends the pre-built receive command to the radio core. This sets up the
+ * radio to receive packets according to the settings in the global rx command.
+ *
+ * @note This function does not alter any of the parameters of the rx command.
+ * It is only concerned with sending the command to the radio core. See @ref
+ * otPlatRadioSetPanId for an example of how the rx settings are set changed.
+ *
+ * @return The value from the command status register.
+ * @retval CMDSTA_Done The command completed correctly.
+ */
+static uint_fast8_t rfCoreSendReceiveCmd(void)
+{
+    sReceiveCmd.status = IDLE;
+    return (RFCDoorbellSendTo((uint32_t)&sReceiveCmd) & 0xFF);
+}
+
+static uint_fast8_t rfCoreSendEdScanCmd(uint8_t aChannel, uint16_t aDurration)
+{
+    static const rfc_CMD_IEEE_ED_SCAN_t cEdScanCmd =
+    {
+        .commandNo                  = CMD_IEEE_ED_SCAN,
+        .startTrigger               =
+        {
+            .triggerType            = TRIG_NOW,
+        },
+        .condition                  = {
+            .rule                   = COND_NEVER,
+        },
+        .ccaOpt                     =
+        {
+            .ccaEnEnergy            = 1,
+            .ccaEnCorr              = 1,
+            .ccaEnSync              = 1,
+            .ccaCorrOp              = 1,
+            .ccaSyncOp              = 0,
+            .ccaCorrThr             = 3,
+        },
+        .ccaRssiThr                 = -90,
+        .endTrigger                 =
+        {
+            .triggerType            = TRIG_REL_START,
+            .pastTrig               = 1,
+        },
+    };
+    sEdScanCmd = cEdScanCmd;
+
+    sEdScanCmd.channel = aChannel;
+
+    /* durration is in ms */
+    sEdScanCmd.endTime = aDurration * (CC2652_RAT_TICKS_PER_SEC / 1000);
+
+    return (RFCDoorbellSendTo((uint32_t)&sEdScanCmd) & 0xFF);
+}
+
+/**
+ * enables the cpe0 and cpe1 radio interrupts
+ *
+ * Enables the @ref IRQ_LAST_COMMAND_DONE and @ref IRQ_LAST_FG_COMMAND_DONE to
+ * be handled by the @ref RFCCPE0IntHandler interrupt handler.
+ */
+static void rfCoreSetupInt(void)
+{
+    bool interruptsWereDisabled;
+
+    /* We are already turned on by the caller, so this should not happen */
+    if (!PRCMRfReady())
+    {
+        return;
```

Comment:
The return should be at the end, you can use the error handling functions to then jump to it